### PR TITLE
fix(autocmds): make sure we are using codelens correctly

### LINKED ISSUE #2574: Codelens refresh fails on InsertLeave
### Problem description

When codelens refresh is executed by InsertLeave autocomand it throws the following exception:

```
Error detected while processing InsertLeave Autocommands for "<buffer=9>":
Error executing lua callback: /usr/local/share/nvim/runtime/lua/vim/lsp/codelens.lua:119: attempt to index field 'range' (a nil value)
stack traceback:
        /usr/local/share/nvim/runtime/lua/vim/lsp/codelens.lua:119: in function </usr/local/share/nvim/runtime/lua/vim/lsp/codelens.lua:113>
```

### LunarVim version

master-aa4c993

### Neovim version (>= 0.7)

NVIM v0.8.0-dev+133-g82c7a82c3

### Operating system/version

macOs 12.3.1

### Affected language servers

Java

### Steps to reproduce

1. Open a Java class
2. Enter insert mode
3. Leave instert mode

### Actual behavior

nvim throws the error mentioned in the description.

### Expected behavior

_No response_

### log and support info

```shell
:messages

Error detected while processing InsertLeave Autocommands for "<buffer=16>":
Error executing lua callback: /usr/local/share/nvim/runtime/lua/vim/lsp/codelens.lua:119: attempt to index field 'range' (a nil value)
stack traceback:
        /usr/local/share/nvim/runtime/lua/vim/lsp/codelens.lua:119: in function </usr/local/share/nvim/runtime/lua/vim/lsp/codelens.lua:113>
E5108: Error executing lua vim/shared.lua:0: t: expected table, got nil
stack traceback:
        [C]: in function 'error'
        vim/shared.lua: in function 'validate'
        vim/shared.lua: in function <vim/shared.lua:0>
        vim/shared.lua: in function 'tbl_filter'
        ...ladera/.local/share/lunarvim/lvim/lua/lvim/core/info.lua:115: in function 'make_auto_lsp_info'
        ...ladera/.local/share/lunarvim/lvim/lua/lvim/core/info.lua:159: in function 'toggle_popup'
        [string ":lua"]:1: in main chunk
Error detected while processing InsertLeave Autocommands for "<buffer=16>":
Error executing lua callback: /usr/local/share/nvim/runtime/lua/vim/lsp/codelens.lua:119: attempt to index field 'range' (a nil value)
stack traceback:
        /usr/local/share/nvim/runtime/lua/vim/lsp/codelens.lua:119: in function </usr/local/share/nvim/runtime/lua/vim/lsp/codelens.lua:113>
```
```


### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by abzcoding**
# Description
what the title says

Fixes #2574

### CODE REVIEW

**Comment by abzcoding** [comment]
@cladera can you test this out?

**Review by cladera — CHANGES_REQUESTED** [review]

File: lua/lvim/lsp/utils.lua [hunk 1]
```diff
@@ -125,12 +125,16 @@ function M.setup_codelens_refresh(client, bufnr)
...
   })
   vim.api.nvim_create_autocmd("InsertLeave", {
     group = "lsp_code_lens_refresh",
     buffer = bufnr,
-    callback = vim.lsp.codelens.display,
+    callback = function()
+      vim.lsp.codelens.display()
+    end,
   })
 end
```
[the remark this answers is not in the document]
Comment (abzcoding) [reply lua/lvim/lsp/utils.lua:139]: that makes sense 👍

File: lua/lvim/lsp/utils.lua [hunk 2]
```diff
@@ -122,15 +122,12 @@ function M.setup_codelens_refresh(client, bufnr)
...
     buffer = bufnr,
-    callback = vim.lsp.codelens.refresh,
-  })
-  vim.api.nvim_create_autocmd("InsertLeave", {
-    group = "lsp_code_lens_refresh",
-    buffer = bufnr,
-    callback = vim.lsp.codelens.display,
+    callback = function()
+      vim.lsp.codelens.refresh()
+    end,
```
[the remark this answers is not in the document]
Comment (cladera) [reply lua/lvim/lsp/utils.lua:130]: I was going to suggest the same

**Review by kylo252 — APPROVED** [review]
LGTM! 🚀

**Pull request merged**